src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+            .orElse(new Date().toString()));
+      } catch (java.text.ParseException e) {
+        System.out.println("Error parsing created date for libraries for sample " + stock.getId());
+      }
+      if (runs.isEmpty() && maxLibCreated.toInstant().isBefore(TWO_DAYS_AGO)) {
+        return Integer.toString(libraries.size());
+      } else {
+        return "";
+      }
+    }
+
+    public String getNumLanesRunning() {
+      Set<RunDto> running = runs.stream().filter(run -> "Running".equals(run.getState())).collect(Collectors.toSet());
+      Integer lanesRunning = 0;
+      for (Integer runId : lanesForRun.keySet()) {
+        if (running.stream().map(run -> run.getId()).collect(Collectors.toSet()).contains(runId)) {
+          lanesRunning += lanesForRun.get(runId).size();
+        }
+      }
+      return lanesRunning.equals(0) ? "" : lanesRunning.toString();
+    }
+
+    public String getNumLanesCompleted() {
+      Set<RunDto> completed = runs.stream().filter(run -> "Completed".equals(run.getState())).collect(Collectors.toSet());
+      Integer lanesCompleted = 0;
+      for (Integer runId : lanesForRun.keySet()) {
+        if (completed.stream().map(run -> run.getId()).collect(Collectors.toSet()).contains(runId)) {
+          lanesCompleted += lanesForRun.get(runId).size();
+        }
+      }
+      return lanesCompleted.equals(0) ? "" : lanesCompleted.toString();
+    }
+
+    public String getDaysInLab(String end) {
+      Date receivedByLab;
+      Date runCompleted;
+      Instant reportingEnd;
+      try {
+        receivedByLab = getDateTimeFormat().parse(stock.getCreatedDate());
+        if (end != null) {
+          reportingEnd = getDateTimeFormat().parse(end).toInstant();
+        } else {
+          reportingEnd = TODAY;
+        }
+      } catch (java.text.ParseException e) {
+        System.out.println("Error parsing created date for stock " + stock.getId());
+        return "";
+      }
+      long daysSinceReceived = Duration.between(receivedByLab.toInstant(), reportingEnd).toDays();
+      if (runs.isEmpty() || !runs.stream().filter(run -> "Running".equals(run.getState())).collect(Collectors.toList()).isEmpty()) {
+        return Long.toString(daysSinceReceived);
+      } else {
+        RunDto mostRecent = runs.stream().max(Comparator.comparing(RunDto::getCompletionDate)).orElse(null);
+        if (mostRecent == null) return Long.toString(daysSinceReceived);
+        try {
+          runCompleted = getDateTimeFormat().parse(mostRecent.getCompletionDate());
+        } catch (java.text.ParseException e) {
+          System.out.println("Error parsing completion date for run " + mostRecent.getId());
+          return "";
+        }
+        long daysReceivedToCompleted = Duration.between(receivedByLab.toInstant(), runCompleted.toInstant()).toDays();
+        return Long.toString(daysReceivedToCompleted);
+      }
+    }
+
+    @Override
+    public int compareTo(ReportItem other) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1210)